### DISCUSSION TITLE
Custom collider

### DISCUSSION THREAD

**Discussion by fhoenig** [jrouwe/JoltPhysics#393] [General]
Hey there!

I've been looking at integrating Jolt into our engine for a while now and I'm finally getting to the point where I can start.
Is there a quick jump start / primer someone could point me to on how do a custom collider type. My models/bodies are signed distance functions, algebraic (slow) and it sparse discrete form. Could pack a sphere tree into it also.

Quite broad of a question I know :-)

**Comment by jrouwe** [comment]
Hello,

There wasn't any documentation about this yet, but I've added a new section 'Creating Custom Shapes' to [the documentation](https://jrouwe.github.io/JoltPhysics/).

Let me know if this is sufficient.

Jorrit.

**Comment by fhoenig** [comment]
Ah cool. Thanks.

Btw, if I wind up packing spheres inside the negative space of the SDF, do you think CompoundShape of your native spheres would perform and if so, how much? As in, what amount of spheres are reasonable?

**Comment by jrouwe** [comment]
I have never tried a sphere tree, so this is a guess:

If you use a StaticCompoundShape I don't think the amount matters a lot, I think it will scale mostly with the amount of spheres that touch a dynamic object at the same time. I would aim for something in the order of 10 contacts between a dynamic object and the sphere tree so if your average object size is 10x10x10 cm your spheres should probably not have a radius that's smaller than 3 cm (object would rest on 3x3 spheres).

Note that there is a limit to the amount of sub shapes you can push in a StaticCompoundShape, the limiting factor for a sphere tree will probably be that the (quad) tree cannot have more than 2 million nodes (the other limit is the amount of bits in the sub shape ID, but I think you'll hit that later for a sphere tree).
